src/dmd/cppmanglewin.d

```diff
@@ -897,10 +897,33 @@ private:
             return;
         }
 
+        void writeName(const(char)[] name)
+        {
+            assert(name);
+            if (is_dmc_template)
+            {
+                if (checkAndSaveIdent(name))
+                return;
+            }
+            else
+            {
+                if (dont_use_back_reference)
+                {
+                    saveIdent(name);
+                }
+                else
+                {
+                    if (checkAndSaveIdent(name))
+                    return;
+                }
+            }
+            buf.writestring(name);
+            buf.writeByte('@');
+        }
         if (TemplateInstance ti = sym.isTemplateInstance())
         {
             auto id = ti.tempdecl.ident;
-            const(char)* symName = id.toChars();
+            auto symName = id.toString();
```

Comment:
It appears that `symName` is *always* an `Identifier` and `saved_idents` are *always* `Identifiers`. If true, all of these strings can be replaced with `Identifier` and then one never has to use the string comparisons.

Reply:
I'm doing this reactor for https://github.com/dlang/dmd/pull/11835 in which mangle overrides are strings not idents.

Reply:
Actually this can be made to work.